src/services/logger.ts

```diff
@@ -18,49 +18,127 @@ if (config.nodeEnv !== 'dev' && config.nodeEnv !== 'test') {
   });
 }
 
-const logger = config.nodeEnv === 'dev' || config.nodeEnv === 'test' ? console : log4js.getLogger();
+const devEnv = config.nodeEnv === 'dev' || config.nodeEnv === 'test';
+const logger = devEnv ? console : log4js.getLogger();
+
+const createTimeStamp = (): string => {
+  const dateObj = new Date();
+
+  let date = dateObj.toISOString();
+  date = date.substring(0, date.indexOf('T'));
+
+  const time = dateObj.toLocaleTimeString([], { hour12: false });
+
+  return `${date} ${time}`;
+};
 
 export class LoggerService {
+  /*
+   * Internal fields are called by the class when each respective method is called
+   *
+   * Each field is by default `null`, see `constructor()` for how each log level is set */
+  #info: (message: string, serviceOperation?: string) => void = () => null;
+  #debug: (message: string, serviceOperation?: string) => void = () => null;
+  #trace: (message: string, serviceOperation?: string) => void = () => null;
+  #warn: (message: string, serviceOperation?: string) => void = () => null;
+  #error: (message: string | Error, innerError?: unknown, serviceOperation?: string) => void = () => null;
+  constructor() {
+    // enable logging only if we're on a dev environment or apm logging is activated
+    if (config.apmLogging || devEnv) {
```

Comment:
nope, 2 things - apmLogging should only be used for APM, not logging. Furthermore, we've got loglevels to disable logging (eg, logLevel.OFF). Secondly, if it's devEnv, logger should be Console. 

This entire IF can be removed, and the below line kept:
`const logger = config.nodeEnv === 'dev' || config.nodeEnv === 'test' ? console : log4js.getLogger();`